appcli/backup_manager/remote_strategy.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Remote backup strategy classes.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+# standard libraries
+import os
+import urllib.parse
+from pathlib import Path
+
+# vendor libraries
+import boto3
+from botocore.exceptions import ClientError
+
+# local libraries
+from appcli.crypto.cipher import Cipher
+from appcli.logger import logger
+
+
+class RemoteStrategy:
+    """
+    Base class for a remote backup strategy.
+    """
+
+    # --------------------------------------------------------------------------
+    # CONSTRUCTOR
+    # --------------------------------------------------------------------------
+    def __init__(self, conf, key_file: Path):
+        self.name = conf["name"]
+        self.type = conf["type"]
+        self.frequency = conf["frequency"]
+        self.configuration = conf["configuration"]
+        self.key_file = key_file
+
+    # ------------------------------------------------------------------------------
+    # PUBLIC METHODS
+    # ------------------------------------------------------------------------------
+    def backup(self, backup_filename):
+        pass
+
+
+class AwsS3Strategy(RemoteStrategy):
+    """
+    Remote backup strategy for pushing a backup to an S3 bucket.
+    Implements RemoteStrategy.
+    """
+
+    # --------------------------------------------------------------------------
+    # CONSTRUCTOR
+    # --------------------------------------------------------------------------
+    def __init__(self, conf, key_file: Path):
+        super().__init__(conf, key_file)
+
+        cipher = Cipher(self.key_file)
+
+        self.s3_bucket = self.configuration["bucket_name"]
+        # Both the access_key and secret_key should be encrypted, decrypt them.
+        self.s3_access_key = cipher.decrypt(self.configuration["access_key"])
+        self.s3_secret_key = cipher.decrypt(self.configuration["secret_key"])
+
+        self.s3_bucket_path = self.configuration["s3_bucket_path"]
+        self.s3_tags = self.configuration["tags"]
+
+    # --------------------------------------------------------------------------
+    # OVERRIDE: RemoteStrategy
+    # --------------------------------------------------------------------------
+
+    def backup(self, backup_filename):
+        logger.info(f"Initiating the S3 backup '{self.name}'.")
+        logger.info(f"Bucket Name '{self.s3_bucket}'.")
+        logger.info(f"Bucket access_key '{self.s3_access_key}'.")
+        logger.info(f"Bucket path '{self.s3_bucket_path}'.")
```

Comment:
More a suggestion, but I reckon printing out these values would look really good with `tabulate` like we do in `cli_builder.py` : https://github.com/brightsparklabs/appcli/blob/develop/appcli/cli_builder.py#L203

Please also log the `backup_filename` and `self.s3_tags`